### PR TITLE
Blacklist refactor and bug fixes

### PR DESCRIPTION
**Pull request by anthonymartin**
- fixes blacklist feature (won't follow or add comments if blacklist entry exists)
- refactors blacklist
- blacklist class created
- blacklist unit tests

### CODE REVIEW

**Comment by timgrossmann** [comment]
@anthonymartin Could you please check if you can make the ci succeed and fix the codacy issues? 😊

**Comment by anthonymartin** [comment]
Do you know if I put an underscore _ for the variable that's not used will pass the linter? https://app.codacy.com/app/timgrossmann/InstaPy/file/36015097899/issues/source?bid=11848437&fileBranchId=13282535#l34

and do you or anyone else have suggestions for how to handle this issue? 
https://app.codacy.com/app/timgrossmann/InstaPy/file/36015097899/issues/source?bid=11848437&fileBranchId=13282535#l14

I need to instantiate the InstaPy class to make my test run, but I don't need it to actually connect to instagram

**Comment by breuerfelix** [comment]
> Do you know if I put an underscore _ for the variable that's not used will pass the linter? https://app.codacy.com/app/timgrossmann/InstaPy/file/36015097899/issues/source?bid=11848437&fileBranchId=13282535#l34
>
 
Yes that works :). 
according to [this](https://stackoverflow.com/questions/24928585/python-for-loop-without-index-and-item/24928645) SO post that works.

> and do you or anyone else have suggestions for how to handle this issue?
> https://app.codacy.com/app/timgrossmann/InstaPy/file/36015097899/issues/source?bid=11848437&fileBranchId=13282535#l14
> 
> I need to instantiate the InstaPy class to make my test run, but I don't need it to actually connect to instagram

We gonna disable this warning because we also have this warning in the main repo.

Codacy is not as important as Travis :) Do you know why you have this many travis Errors ?  
3.7 is running fine but all other version are getting alot of errors. 

btw awesome work !! this is really awesome :)

File: tests/blacklist_test.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+class TestBlacklist(unittest.TestCase):
+    """Update user name and password below. Tests run in PyCharm CE, but I can't seem to get them working from the command line at the moment"""
+    @classmethod
+    def setUpClass(cls):
+        username = ''
+        password = ''
+        cls.test_dir =  "{}/_instapy_test".format(file_manager.get_home_path())
+        cls.test_dir = file_manager.slashen(cls.test_dir, 'native')
+        set_workspace(path=cls.test_dir)
+        cls.session = InstaPy(username=username, password=password, bypass_suspicious_attempt=True, headless_browser=True)
```
Comment (breuerfelix) [inline tests/blacklist_test.py:18]: you dont need to create an instapy class here, I guess if you would remove this, all tests would run fine again.

you don't reuse the session anyways in your tests so it doesn't matter :)

**Review by breuerfelix — CHANGES_REQUESTED** [review]

**Pull request closed without merging**